x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -365,8 +365,16 @@ const serviceAgentRoute = createApmServerRoute({
   handler: async (
     resources
   ): Promise<
-    | { agentName?: undefined; runtimeName?: undefined }
-    | { agentName: string | undefined; runtimeName: string | undefined }
+    | {
+        agentName?: undefined;
+        runtimeName?: undefined;
+        cloudProviderAndService?: undefined;
+      }
+    | {
+        agentName: string | undefined;
+        runtimeName: string | undefined;
+        cloudProviderAndService: string | undefined;
+      }
```

Comment:
Again I see this was already the case but these two options look like the same.

What happens if you change it to:
```suggestion
  {
        agentName?: string;
        runtimeName?: string;
        cloudProviderAndService?: string ;
      }
```